src/main/java/movie/storage/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,33 @@
+package movie.storage.service.impl;
+
+import java.util.List;
+import movie.storage.dao.OrderDao;
+import movie.storage.lib.Inject;
+import movie.storage.lib.Service;
+import movie.storage.model.Order;
+import movie.storage.model.Ticket;
+import movie.storage.model.User;
+import movie.storage.service.OrderService;
+import movie.storage.service.ShoppingCartService;
+
+@Service
+public class OrderServiceImpl implements OrderService {
+    @Inject
+    private OrderDao orderDao;
+    @Inject
+    private ShoppingCartService shoppingCartService;
+
+    @Override
+    public Order completeOrder(List<Ticket> tickets, User user) {
+        Order order = new Order();
+        order.setUser(user);
+        order.setTickets(tickets);
```

Comment:
orderDao.add() will also create a delete statement for shoppingCart tickets(as in shoppingCartService.clear). Let's use deep cloning here

Reply:
Done